designs/credentials/credentials-management.md

```diff
@@ -0,0 +1,65 @@
+# Credentials Management
+
+This outlines how the Toolkit produces and obtains credentials.
+
+## Terminology
+
+### Credentials
+
+Credentials allow the Toolkit to interact with AWS on a user's behalf. Service clients require a [Credentials](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Credentials.html) object, and the Toolkit obtains these objects through Credentials Providers.
+
+### Credentials Provider
+
+Credentials Providers are how the toolkit abstracts away different ways of obtaining a user's credentials, and produces Credentials objects. For example, a Shared Credentials Provider knows how to obtain credentials for a specific profile within [Shared Credentials Files](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/loading-node-credentials-shared.html).
+
+A Credentials Provider is produced by a Credentials Provider Factory
+
+### Credentials Provider Factory
+
+A factory is capable of producing one or more Credentials Providers for a single credentials type. For example, a Shared Credentials related factory produces Credentials Providers for each profile found within shared credentials files.
+
+### Credentials Provider Id
```

Comment:
nit: this is a weird name because it's made up of what I would think of as the credentials provider id (the 'profile' is the id of the credentials provider 'shared credentials provider')

Reply:
I've updated the section to indicate that the Id applies to CredentialsProvider _instances_. Let me know if this feels better?